src/core/payoutCalculator/PayoutCalculatorIsolateSuperCharge.ts

```diff
@@ -43,11 +53,22 @@ export class PayoutCalculatorIsolateSuperCharge implements IPayoutCalculator {
                     [key: string]: { npsStake: number; commonStake: number; superchargedStake: number };
                 } = {};
 
-                const totalRewards = block.coinbase + block.feetransfertoreceiver - block.feetransferfromcoinbase;
-                const totalNPSPoolRewards = stakeIsLocked(winner, block) ? block.coinbase : block.coinbase / 2;
-                const totalSuperchargedPoolRewards = stakeIsLocked(winner, block) ? 0 : block.coinbase / 2;
-                const totalCommonPoolRewards = totalRewards - totalNPSPoolRewards - totalSuperchargedPoolRewards;
+                let totalRewards = coinbase + block.feetransfertoreceiver - block.feetransferfromcoinbase;
+                let totalNPSPoolRewards = winnerStakeIsLocked ? coinbase : coinbase / 2;
+                let totalSuperchargedPoolRewards = winnerStakeIsLocked ? 0 : coinbase / 2;
+                
+                let totalCommonPoolRewards = totalRewards - totalNPSPoolRewards - totalSuperchargedPoolRewards;
+                let blockRewardsToBurn = 0;
+                
+                if (burnScr) {
+                    totalRewards = (coinbase / 2) + block.feetransfertoreceiver - block.feetransferfromcoinbase;
+                    totalNPSPoolRewards = coinbase / 2;
+                    totalSuperchargedPoolRewards = coinbase / 2;
```

Comment:
This should be zero if burning? Otherwise totalCommonPoolRewards will be negative?